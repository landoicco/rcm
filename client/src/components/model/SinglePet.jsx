//@flow

import React from "react";
import useCatImageUrl from "../../hooks/useCatImageUrl";
import type { DataItem } from "../../commons/types";

const SinglePet = ({ item }: DataItem): React.Node => {
  const catImageUrl = useCatImageUrl();
  return (
    <div className="row">
      <div className="col s12 m7">
        <div className="card">
          <div className="row">
            <div className="card-image col s4">
              <img
                src={catImageUrl}
                style={{ width: "200px", height: "200px" }}
                alt="cat or dog"
              />
            </div>
            <div className="card-content col offset-s1 s7">
              <p>
                This is a simple description of the pet in question **TODO**
              </p>
              <p style={{ marginTop: "50px" }}>
                <b style={{ color: "#2B874C" }}>Address:</b> {item.address}
              </p>
            </div>
          </div>
          <div className="card-action">
            <p href="/">
              This is a {item.type} named{" "}
              <b style={{ color: "#2B874C" }}>{item.name}</b>
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SinglePet;
