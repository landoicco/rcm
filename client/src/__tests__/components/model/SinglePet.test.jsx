import { render, screen } from "@testing-library/react";
import SinglePet from "../../../components/model/SinglePet";

const mockPet = {
  id: "-1",
  name: "Testy",
  address: "here",
  type: "Dog",
};

describe("SinglePet", () => {
  test("The component render correctly", () => {
    const { container } = render(<SinglePet item={mockPet} />);
    expect(container).toMatchSnapshot();
  });

  test("The component shows the pet name", async () => {
    render(<SinglePet item={mockPet} />);
    const element = await screen.findByText(mockPet.name);
    expect(element).toBeDefined();
  });

  test("The component shows the pet address", async () => {
    render(<SinglePet item={mockPet} />);
    const element = await screen.findByText(mockPet.address);
    expect(element).toBeDefined();
  });

  // TODO: Implement...
  test("The component shows the pet type", () => {});
});
