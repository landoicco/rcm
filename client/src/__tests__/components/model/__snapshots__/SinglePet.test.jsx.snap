// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SinglePet The component render correctly 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col s12 m7"
    >
      <div
        class="card"
      >
        <div
          class="row"
        >
          <div
            class="card-image col s4"
          >
            <img
              alt="cat or dog"
              src=""
              style="width: 200px; height: 200px;"
            />
          </div>
          <div
            class="card-content col offset-s1 s7"
          >
            <p>
              This is a simple description of the pet in question **TODO**
            </p>
            <p
              style="margin-top: 50px;"
            >
              <b
                style="color: rgb(43, 135, 76);"
              >
                Address:
              </b>
               
              here
            </p>
          </div>
        </div>
        <div
          class="card-action"
        >
          <p
            href="/"
          >
            This is a 
            Dog
             named
             
            <b
              style="color: rgb(43, 135, 76);"
            >
              Testy
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
